src/gl/gl_builders.js

```diff
@@ -145,213 +145,149 @@ GLBuilders.buildPolylines = function (
-        var line_debug = null;
-        if (intersection != null) {
-            var intersect_outer = intersection;
-
-            // Cap the intersection point to a reasonable distance (as join angle becomes sharper, miter joint distance would approach infinity)
-            var len_sq = Vector.lengthSq([intersect_outer[0] - joint[0], intersect_outer[1] - joint[1]]);
-            var miter_len_max = 3; // multiplier on line width for max distance miter join can be from joint
-            if (len_sq > (width * width * miter_len_max * miter_len_max)) {
-                line_debug = 'distance';
-                intersect_outer = Vector.normalize([intersect_outer[0] - joint[0], intersect_outer[1] - joint[1]]);
-                intersect_outer = [
-                    joint[0] + intersect_outer[0] * miter_len_max,
-                    joint[1] + intersect_outer[1] * miter_len_max
-                ];
-            }
+        // Do this with the rest (except the last one)
+        for(i = 1; i < lineSize - 1 ; i++){
+            prevCoord = currCoord;
+            currCoord = nextCoord;
+            nextCoord = line[i+1];
 
-            var intersect_inner = [
-                (joint[0] - intersect_outer[0]) + joint[0],
-                (joint[1] - intersect_outer[1]) + joint[1]
-            ];
+            //  a) Store prev normal and compute the next one
+            normPrevCurr = normCurrNext;
+            normCurrNext = Vector.normalize( Vector.perp( currCoord, nextCoord ) );
+            rightNorm = Vector.normalize( Vector.add(normPrevCurr, normCurrNext) );
 
-            vertices.push(
-                intersect_inner, intersect_outer, pa_inner[0],
-                pa_inner[0], intersect_outer, pa_outer[0],
+            var scale = Math.sqrt(2 / (1 + Vector.dot(normPrevCurr,normCurrNext)));
+            rightNorm = Vector.mult(rightNorm,scale);
+            //  TODO:
+            //          - compare to the miterlimit
+            //          - if bigger 
 
-                pb_inner[1], pb_outer[1], intersect_inner,
-                intersect_inner, pb_outer[1], intersect_outer
-            );
+            addVertexPair(currCoord, rightNorm, i/lineSize );
         }
-        else {
-            // Line segments are parallel, use the first outer line segment as join instead
-            line_debug = 'parallel';
-            pa_inner[1] = pb_inner[0];
-            pa_outer[1] = pb_outer[0];
-
-            vertices.push(
-                pa_inner[1], pa_outer[1], pa_inner[0],
-                pa_inner[0], pa_outer[1], pa_outer[0],
-
-                pb_inner[1], pb_outer[1], pb_inner[0],
-                pb_inner[0], pb_outer[1], pb_outer[0]
-            );
+
+        rightNorm = Vector.normalize( normCurrNext );
+        addVertexPair(nextCoord, rightNorm, 1 );
+
+        // Add vertices to buffer acording their index
+        for (i = 0; i < lineSize - 1; i++) {
```

Comment:
Yeah I agree that we should not use features that could cause tangram to be slow. I looked into how they implemented `let` in traceur. For the following code, 
```javascript
let a = [1, 2, 3, 4];
let i = 10;
for (let i in a) {
    console.log(i);
}

function do_something() {
  console.log(foo); // ReferenceError
  let foo = 2;
}

if (true) {
    let x = 'a';
    console.log(x);
    if (true) {
        let x = 'b';
        console.log(x);
        if (true) {
            let x = 'c';
            console.log(x);
            if (true) {
                let x = 'd';
                console.log(x);
            }
        }
    }
}
```
traceur generates 
```javascript
System.registerModule("test.js", [], function() {
  "use strict";
  var __moduleName = "test.js";
  var a = [1, 2, 3, 4];
  var i = 10;
  for (var i$__0 in a) {
    console.log(i$__0);
  }
  function do_something() {
    console.log(foo);
    var foo = 2;
  }
  if (true) {
    var x = 'a';
    console.log(x);
    if (true) {
      var x$__1 = 'b';
      console.log(x$__1);
      if (true) {
        var x$__2 = 'c';
        console.log(x$__2);
        if (true) {
          var x$__3 = 'd';
          console.log(x$__3);
        }
      }
    }
  }
  return {};
});
System.get("test.js" + '');
```
I need to make sure that traceur always uses this method for compiling `let`, but if so then it should be no or minimal overhead.
 
I can look into this more tomorrow as there might be situations where they need try and catch.